packages/pluggableWidgets/barcode-scanner-web/src/hooks/useCodeScanner.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import { BarcodeFormat, BrowserMultiFormatReader, DecodeHintType } from "@zxing/library";
+
+const hints = new Map();
+// `BarcodeFormat` is a TypeScript enum. Calling `Object.values` on it returns an array of string and ints, we only want the latter.
+const formats = Object.values(BarcodeFormat).filter(format => typeof format !== "string") as BarcodeFormat[];
+hints.set(DecodeHintType.POSSIBLE_FORMATS, formats);
```

Comment:
The creating of map here looks unnecessary as you are using it only in line 27 and with one key. Just inline there..

Reply:
The library requires a map and I didn't want to inline it because we only need to create it once. But I'm fine with it either way.